client/ayon_core/pipeline/create/product_name.py

```diff
@@ -58,14 +57,16 @@ def get_product_name_template(
         "task_types": task_type
     }
 
-    if is_product_base_type_supported():
-        if product_base_type:
-            filtering_criteria["product_base_types"] = product_base_type
-        else:
-            warn(
-                "Product base type is not provided, please update your"
-                "creation code to include it. It will be required in "
-                "the future.", DeprecationWarning, stacklevel=2)
+    if not product_base_type:
+        warn(
+            "Product base type is not provided, please update your"
+            "creation code to include it. It will be required in "
+            "the future.",
+            DeprecationWarning,
+            stacklevel=2
+        )
+    filtering_criteria["product_base_types"] = product_base_type
```

Comment:
```suggestion
        "task_types": task_type,
        "product_base_types": product_base_type,
    }
    if not product_base_type:
        warn(
            "Product base type is not provided, please update your"
            "creation code to include it. It will be required in "
            "the future.",
            DeprecationWarning,
            stacklevel=2
        )